src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -1796,6 +1895,34 @@ Containers::Optional<AnimationData> UfbxImporter::doAnimation(UnsignedInt id) {
     return AnimationData{std::move(data), std::move(tracks)};
 }
 
+MAGNUM_UFBXIMPORTER_LOCAL UnsignedInt UfbxImporter::doSkin3DCount() const {
+    return UnsignedInt(_state->scene->skin_deformers.count);
+}
+
+MAGNUM_UFBXIMPORTER_LOCAL Int UfbxImporter::doSkin3DForName(Containers::StringView name) {
+    return typedId(ufbx_find_element_len(_state->scene.get(), UFBX_ELEMENT_SKIN_DEFORMER, name.data(), name.size()));
+}
+
+MAGNUM_UFBXIMPORTER_LOCAL Containers::String UfbxImporter::doSkin3DName(UnsignedInt id) {
+    return _state->scene->skin_deformers[id]->name;
+}
+
+MAGNUM_UFBXIMPORTER_LOCAL Containers::Optional<SkinData3D> UfbxImporter::doSkin3D(UnsignedInt id) {
+    const ufbx_scene* scene = _state->scene.get();
+    const ufbx_skin_deformer* skin = scene->skin_deformers[id];
+
+    Containers::Array<UnsignedInt> joints{NoInit, skin->clusters.count};
+    Containers::Array<Matrix4> inverseBindMatrices{ValueInit, skin->clusters.count};
```

Comment:
This feels like it could be `NoInit` but I ran into allocator issues if so. I assume that it's because the default allocator (required for ABI crossing) is the C++ `new[]/delete[]` which conflates allocation and initialization?

Reply:
It said that "can't use a non-default deleter", right? Yeah, that's it. The default is `new`/`delete` to correctly handle types that need construction/destruction, and anything else needs a deleter pointer (which then may point to the already-unloaded plugin binary in some cases, which is undesired, so it's outright banned).

And `NoInit` can't be easily implemented on non-trivially-constructible types, at least until C++ adds yet another `new` overload that allows me to tell it to not construct, so it's done as `new char[]`, `reinterpret_cast`, and a custom deleter that casts back before passing the pointer to `delete[]`. In C++20 we got crap like `constexpr` allocation that has little practical use, but the *actual issues* get never solved because they're deemed too boring for the commitee or something. Sigh.

Reply:
Yea that's exactly the issue, not a massive performance hit but goes into the pile of C++ annoyances.